### PR TITLE
fix(cli): ensure `cliui.Agent` doesn't fetch infinitely

### LINKED ISSUE #8426: cli: ssh: open SSH connection keeps hitting /api/v2/workspaceagents/{workspaceagent}
# Problem:

When connecting to a workspace over SSH, we run the [following code](https://github.com/coder/coder/blob/75f62dc39dd218870e7f7d3be009f142df82a156/cli/ssh.go#L180) to show a spinny indicator. If not otherwise specified, `AgentOptions.FetchInterval` defaults to [500ms](https://github.com/coder/coder/blob/75f62dc39dd218870e7f7d3be009f142df82a156/cli/cliui/agent.go#L26). 

This never stops running, meaning that 100 workspace connections over SSH equate to 200 requests to `/api/v2/workspaceagents/{workspaceagent}` per second.

# Proposed solution

Once we're connected to the workspace we probably don't need to continue fetching the agent. Just stop the ticker once the connection is established.

### PR DESCRIPTION
**Pull request by coadler**
Fixes: https://github.com/coder/coder/issues/8426

### CODE REVIEW

File: cli/cliui/agent_test.go [hunk 1]
```diff
@@ -350,4 +348,37 @@ func TestAgent(t *testing.T) {
...
+				require.Never(t, func() bool {
+					called := atomic.LoadUint64(&fetchCalled)
+					return called > 5 || called == 0
+				}, time.Second, 100*time.Millisecond)
+
+				return nil
+			},
+		}
+		require.NoError(t, cmd.Invoke().Run())
+	})
```
Comment (johnstcn) [inline cli/cliui/agent_test.go:383]: :chefs-kiss:

**Review by johnstcn — APPROVED** [review]

File: cli/cliui/agent.go [hunk 2]
```diff
@@ -15,13 +15,16 @@ var errAgentShuttingDown = xerrors.New("agent is shutting down")
 
 type AgentOptions struct {
 	FetchInterval time.Duration
-	Fetch         func(context.Context) (codersdk.WorkspaceAgent, error)
+	Fetch         func(ctx context.Context, agentID uuid.UUID) (codersdk.WorkspaceAgent, error)
```
Comment (mafredri) [inline cli/cliui/agent.go:18]: I'm glad you made this change too (to pass agent ID)!

**Review by mafredri — APPROVED** [review]
Nice catch, thanks for fixing this!

**Comment by johnstcn** [comment]
Can confirm that starting lots of parallel `coder ssh` invocations no longer kills the database.

**Pull request merged**